model/src/main/java/uk/gov/hmcts/fees2/register/data/service/impl/IdamServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+package uk.gov.hmcts.fees2.register.data.service.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import uk.gov.hmcts.fees2.register.data.exceptions.GatewayTimeoutException;
+import uk.gov.hmcts.fees2.register.data.exceptions.UserNotFoundException;
+import uk.gov.hmcts.fees2.register.data.model.IdamUserIdResponse;
+import uk.gov.hmcts.fees2.register.data.service.IdamService;
+
+import java.lang.reflect.Array;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+public class IdamServiceImpl implements IdamService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(IdamServiceImpl.class);
+
+    private static final String USERID_ENDPOINT = "/api/v1/users";
+
+    @Value("${auth.idam.client.baseUrl}")
+    private String idamBaseURL;
+
+    @Autowired()
+    @Qualifier("restTemplateIdam")
+    private RestTemplate restTemplateIdam;
+
+    @Override
+    public String getUserName(final MultiValueMap<String, String> headers, final String userId) {
+        LOG.info("userID : {}", userId);
+
+        try {
+
+            final ResponseEntity<IdamUserIdResponse[]> responseEntity = getResponseEntity(headers, userId);
+
+            if (null != responseEntity && null != responseEntity.getBody()) {
+
+                final IdamUserIdResponse[] idamUserIdResponse = responseEntity.getBody();
+                LOG.info("idamUserIdResponse: {}", idamUserIdResponse);
+
+                if (idamUserIdResponse != null && idamUserIdResponse.length >= 1) {
+
+                    IdamUserIdResponse response = (IdamUserIdResponse) Array.get(idamUserIdResponse, 0);
+
```

Comment:
As discussed, Pls add a null check for a response and modify info level logs

Reply:
Done